src/image/loading_displaying.js

```diff
@@ -646,7 +646,20 @@ function _createGif(
   failureCallback,
   finishCallback
 ) {
-  const gifReader = new omggif.GifReader(arrayBuffer);
+  let gifReader;
+  try {
+    gifReader = new omggif.GifReader(arrayBuffer);
+  } catch (e) {
+    p5._friendlyFileLoadError(8, pImg.src);
+    if (typeof failureCallback === 'function') {
+      failureCallback(e);
+    } else {
+      console.error(e);
+    }
+    finishCallback();
```

Comment:
I think for consistency sake with other loading functions we don't call `finishCallback()` here which will resolve the `preload` even in the case of an error.